Source/WebCore/css/StyleProperties.cpp

```diff
@@ -705,6 +707,64 @@ String StyleProperties::fontVariantValue() const
     return result.toString();
 }
 
+String StyleProperties::fontSynthesisValue() const
+{
+    StringBuilder result;
+
+    auto getExplicitLonghandValue = [&](CSSPropertyID propertyID) -> CSSValue* {
+        auto foundPropertyIndex = findPropertyIndex(propertyID);
+        if (foundPropertyIndex == -1)
+            return nullptr;
+
+        auto property = propertyAt(foundPropertyIndex);
+        if (property.isImplicit())
+            return nullptr;
+
+        return property.value();
+    };
+
+    // font-synthesis: none | [ weight || style || small-caps ]
+    auto weightValue = getExplicitLonghandValue(CSSPropertyFontSynthesisWeight);
+    auto styleValue = getExplicitLonghandValue(CSSPropertyFontSynthesisStyle);
+    auto capsValue = getExplicitLonghandValue(CSSPropertyFontSynthesisSmallCaps);
+
+    auto valueID = [&](CSSValue *value) {
+        if (!value || !is<CSSPrimitiveValue>(value))
+            return CSSValueInvalid;
+        return downcast<CSSPrimitiveValue>(value)->valueID();
+    };
+
+    auto weightValueID = valueID(weightValue);
+    auto styleValueID = valueID(styleValue);
+    auto capsValueID = valueID(capsValue);
+
+    if (weightValueID != CSSValueInvalid && weightValueID == styleValueID && weightValueID == capsValueID) {
+        // font-synthesis: value can be none or CSS wide value keyword
```

Comment:
```suggestion
        // Handle `none` or CSS wide-keywords.
```
nit: WebKit comment style starts with uppercase, and ends with `.`